packages/devp2p/src/dpt/kbucket.ts

```diff
@@ -43,7 +43,7 @@ export class KBucket extends EventEmitter {
 
     const keys = []
     if (Buffer.isBuffer(obj.id)) keys.push(obj.id.toString('hex'))
-    //if (obj.address && obj.port) keys.push(`${obj.address}:${obj.port}`)
+    if (obj.address && obj.tcpPort) keys.push(`${obj.address}:${obj.tcpPort}`)
```

Comment:
Not sure why this was commented out (maybe perceived redundancy?). 

Peer descriptions obtained via DNS do not have an `id` (unlike peers from `findNeighbours`) so without synthesizing this ip based key they aren't being banned consistently.

Reply:
I commented this out since there was just no `port` parameter on the object and I wasn't sure if to add the tcp or udp port instead.